addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -2964,7 +2964,11 @@ var SnippetsMenu = Widget.extend({
                     .is('.o_not_editable :not([contenteditable="true"]), .o_not_editable');
             }
             if (isChildren) {
-                return !$(this).is('.o_not_editable *');
+                // it's a drop-near.
+                // Prevents inner blocks from being dropped near elements within
+                // an image field.
+                const childOfImageField = this.closest("[data-oe-type=image]");
+                return !$(this).is('.o_not_editable *') && !childOfImageField;
```

Comment:
No need to search for `!$(this).is('.o_not_editable *')` if the second part of the condition is not true: invert the && operands.